### PR TITLE
[OMNI-1929] Fixing One-Of Required Flag

### PR DESCRIPTION
**Pull request by bc-ninablanson**
# WHAT

Updating to fix instance where one-of object was being marked as `Optional` even if it had `Required` fields.
# WHY

Confusing UI

@deini @rtalvarez @abernard

### CODE REVIEW

File: dist/schema-form.js [hunk 1]
```diff
@@ -1431,7 +1431,7 @@ angular.module('schemaForm').provider('schemaForm',
             var path = options.path.slice();
             path.push(key);
             
-            var required = schema.required && schema.required.indexOf(key) !== -1;
+            var required = value.required && value.required.length !== 0;
```
Comment (rtalvarez) [inline dist/schema-form.js]: `value.required && value.required.length` should be enough

File: src/services/schema-form.js [hunk 2]
```diff
@@ -314,7 +314,7 @@ angular.module('schemaForm').provider('schemaForm',
             var path = options.path.slice();
             path.push(key);
             
-            var required = schema.required && schema.required.indexOf(key) !== -1;
+            var required = value.required && value.required.length !== 0;
```
Comment (rtalvarez) [inline src/services/schema-form.js:317]: `value.required && value.required.length` should be enough

Comment (bc-ninablanson) [reply src/services/schema-form.js:317]: I tried but it kept bombing out and returning the actual length of the array. It was super bizarre.

Comment (rtalvarez) [reply src/services/schema-form.js:317]: Yeah, it will return the actual length of the array, but you can use that as a truthy value

Comment (rtalvarez) [reply src/services/schema-form.js:317]: Its 🍹 anyway, no big deal

**Comment by rtalvarez** [comment]
👍

**Pull request merged**